src/gui/cloudproviders/cloudproviderwrapper.cpp

```diff
@@ -206,6 +212,13 @@ void CloudProviderWrapper::slotSyncFinished(const SyncResult &result)
     updateStatusText(result.statusString());
 }
 
+static GMenuItem* addMenuItem(const QString text, const gchar *action)
+{
+    QFontMetrics fm = QApplication::fontMetrics();
+    preferredTextWidth = MAX (preferredTextWidth, (fm.boundingRect (text)).width ());
```

Comment:
> I arbitrarily chose to use the main menu width as a baseline for how wide the submenu should be (it can be up to twice as wide).

Forgive me if I'm being thick, but I don't see this being set anywhere? 

`preferredTextWidth` is set at the top as 0, and here `MAX` defaults to setting the larger number, which will lead to the same issue this PR is trying to fix (the submenu entries being huge due to the filenames being long)

Reply:
No worries! The next commit uses the preferredTextWidth to ellide the text. The reason this works is because the preferredTextWidth is max of all of the main menu items (i.e. the hard coded items), but NOT the filenames from recent activity.

Reply:
Duh, this is obvious now that I look more carefully at the changes in lines 231-266. Thanks for pointing it out!